File: python3/module_1/main.py
#!/usr/bin/env python3
import re

import glob2

from module_1.StockAnalysis import StockAnalysis


def main():
    # Output single file as original specs
    StockAnalysis('../data/stocks/Mid_Cap/MUTHOOTFIN.csv') \
        .print() \
        .write_csv(   '../data_output/week2/week2-python.csv'  ) \
        .write_stats( '../data_output/week2/week2-python.json' )

    # Output directory tree of all stocks
    for input_csv in glob2.iglob('../data/stocks/**/*.csv'):
        output_csv  = re.sub( '\.\./data/', '../data_output/week2/python3/', input_csv )
        output_json = re.sub( '\.csv', '.json', output_csv )

        StockAnalysis(input_csv) \
            .write_csv(   output_csv  ) \
            .write_stats( output_json )

if __name__ == "__main__":
    main()

File: scala/src/main/scala/module_1/Main.scala
package module_1

import better.files.{File, _}

object Main extends App {

  // Output single file as original specs
  StockAnalysis.main(Array(
    "../data/stocks/Mid_Cap/MUTHOOTFIN.csv",
    "../data_output/module_1/week2-scala.csv",
    "../data_output/module_1/week2-scala.json"
  ))


  // Output directory tree of all stocks
  val data_dir:   File         = file"../data/stocks/"
  val input_csvs: List[String] = data_dir.glob("**/*.csv").map(_.path.toString).toList
  input_csvs.foreach((input_csv: String) => {
    val output_csv  = input_csv.replaceAll("/data/", "/data_output/module_1/scala/")
    val output_json = output_csv.replaceAll(".csv", ".json")

    StockAnalysis.main(Array(input_csv, output_csv, output_json))
  })
}
